src/openforms/appointments/renderer.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import Any, Iterator
+
+from openforms.formio.rendering.nodes import ComponentNodeBase
+from openforms.formio.service import FormioData, format_value
+from openforms.submissions.rendering import Renderer
+
+from .service import get_appointment
+
+
+@dataclass
+class ContactDetailNode(ComponentNodeBase):
+    """
+    Rendering of a contact detail formio component.
+
+    Ideally we'd be able to share more logic with the
+    :class:`openforms.formio.rendering.nodes.ComponentNode`, but that is too tighly
+    coupled with ``FormStep`` and ``SubmissionStep`` Django models at the moment.
+
+    Therefore, this implementation is extremely minimal.
+    """
+
+    data: FormioData
+
+    @property
+    def value(self) -> Any:
```

Comment:
```suggestion
    def value(self) -> JSONValue:
```
If I read `FormioData` correctly.

Reply:
there are some shenanigans happening in the renderer where the json data is essentially converted `to_python` - sometimes. I think I decided to stick to the superclass interface?

Reply:
`ComponentNode` did something different, but those are siblings in the inheritance tree so updated to `JSONValue`